narwhal/network/src/connectivity.rs

```diff
@@ -115,6 +141,87 @@ impl ConnectionMonitor {
 
         self.connection_statuses.insert(peer_id, connection_status);
     }
+
+    // TODO: Replace this with ClosureMetric
+    fn update_quinn_metrics_for_peer(&self, peer_id: &PeerId, stats: &ConnectionStats) {
+        // Update PathStats
+        self.connection_metrics
+            .network_peer_rtt
+            .with_label_values(&[&format!("{peer_id}")])
+            .observe(stats.path.rtt.as_millis() as f64);
```

Comment:
So we absolutely do not want to do this as this has unbounded cardinality and will completely break grafana unless these per connection metrics are only collected for "well known" peers

Reply:
Good call, won't insert any new peers into our list. Will only track metrics for all known peers at startup.